### PR TITLE
Implement linting with ktlint

### PR DESCRIPTION
**Pull request by shannynalayna**
Lints the sawtooth xo app using ktlint from the command line. Also renames the project from ```io.bitwise.sawtooth_xo``` to ```bitwiseio.sawtooth.xo``` and the internal directory ```rest_api``` to ```api``` to comply with package and directory naming conventions which disallow underscores. 

In order to perform linting, download ktlint (can use ```brew install shyiko/ktlint/ktlint```) and then simply run the command ```ktlint``` from the project's source directory or you may specify files as optional arguments with this command. With this PR, there should be no feedback after this command. If there are any issues, the command will return the file location and the linting warning. Can also run ```./gradlew ktlint``` using the gradle build script. Should build successfully.

### CODE REVIEW

**Comment by boydjohnson** [comment]
Is it possible to have ktlint part of the gradle script?

**Comment by shannynalayna** [comment]
@boydjohnson I've tried to get it integrated but have been running into a few issues while actually running it with the gradle script, so still working on that!

File: examples/xo_android_client/app/build.gradle [hunk 1]
```diff
@@ -46,4 +50,23 @@ dependencies {
...
     implementation 'com.android.support:design:28.0.0'
+    ktlint 'com.github.shyiko:ktlint:0.29.0'
+}
+
+task ktlint(type: JavaExec, group: "verification") {
+    description = "Check Kotlin code style."
+    main = "com.github.shyiko.ktlint.Main"
+    classpath = configurations.ktlint
+    args "src/**/*.kt"
+    // to generate report in checkstyle format prepend following args:
```
Comment (boydjohnson) [inline examples/xo_android_client/app/build.gradle]: We shouldn't have commented out code.

**Review by boydjohnson — CHANGES_REQUESTED** [review]
Commented out code.

**Review by boydjohnson — APPROVED** [review]

**Review by dplumb94 — APPROVED** [review]
I had to do `brew install shyiko/ktlint/ktlint` to install. Tested, works!

**Pull request merged**